src/core_write.cpp

```diff
@@ -17,12 +17,16 @@
 
 UniValue ValueFromAmount(const CAmount& amount)
 {
-    bool sign = amount < 0;
-    int64_t n_abs = (sign ? -amount : amount);
-    int64_t quotient = n_abs / COIN;
-    int64_t remainder = n_abs % COIN;
+    int64_t quotient = amount / COIN;
+    if (quotient < 0) {
+        quotient = -quotient;
+    }
+    int64_t remainder = amount % COIN;
+    if (remainder < 0) {
+        remainder = -remainder;
+    }
     return UniValue(UniValue::VNUM,
```

Comment:
Couldn't this just call FormatMoney?
```suggestion
    return UniValue(UniValue::VNUM, FormatMoney(amount));
```

Reply:
Good point. The only difference between them is that `FormatMoney` chops off some trailing zeroes to make the output more human-friendly. That should be OK also for JSON consumption. Now calling `FormatMoney`.

Reply:
Oh, turns out we have quite a few assumptions about the exact formatting in the tests that read `test/util/data/*.json`. Reverting to the original version!

Reply:
Yes, many clients assume the number of zeros is equal to the subdivision. This is why we don't use the human-readable function for the JSON formatting.